azure/Kqlmagic/my_aad_helper.py

```diff
@@ -92,9 +94,26 @@ def __init__(self, kcsb, default_clientid, **options):
         client_id = kcsb.application_client_id or default_clientid
         self._resource = "{0.scheme}://{0.hostname}".format(urlparse(kcsb.data_source))
         token_cache = None
-        isSso = "FALSE" # os.getenv("{0}_ENABLE_SSO".format(Constants.MAGIC_CLASS_NAME.upper()))
-        if (isSso and isSso.upper() == "TRUE"):
-            token_cache = AdalTokenCache()
+        SSO_id_enc_key = os.getenv("{0}_ENABLE_SSO".format(Constants.MAGIC_CLASS_NAME.upper()))
+        key_vals_SSO = Parser.parse_and_get_kv_string(SSO_id_enc_key, {}) if SSO_id_enc_key else {}
```

Comment:
I suggest you move all the code between line 97 and 114 and put as method in adal_toke_cache, and call it from here